packages/pwa-buildpack/lib/WebpackTools/configureWebpack.js

```diff
@@ -0,0 +1,342 @@
+                    oneOf: [
+                        {
+                            test: [paths.src, ...hasFlag('cssModules')],
+                            use: [
+                                'style-loader',
+                                {
+                                    loader: 'css-loader',
+                                    options: {
+                                        localIdentName:
+                                            '[name]-[local]-[hash:base64:3]',
+                                        modules: true
+                                    }
+                                }
+                            ]
+                        },
+                        {
+                            include: /node_modules/,
+                            use: [
+                                'style-loader',
+                                {
+                                    loader: 'css-loader',
+                                    options: {
+                                        modules: false
+                                    }
+                                }
+                            ]
+                        }
+                    ]
+                },
+                {
+                    test: /\.(jpg|svg)$/,
+                    use: [
+                        {
+                            loader: 'file-loader',
+                            options: {
+                                name: '[name]-[hash:base58:3].[ext]',
+                                outputPath: 'static',
+                                publicPath: '/static'
+                            }
+                        }
+                    ]
+                }
+            ]
+        },
+        resolve: await MagentoResolver.configure({
+            paths: {
+                root: context
+            }
+        }),
+        plugins: [
+            new RootComponentsPlugin({
+                rootComponentsDirs: [
+                    ...hasFlag('rootComponents'),
+                    context
+                ].reduce(
+                    (searchPaths, moduleDir) => [
+                        ...searchPaths,
+                        path.join(moduleDir, 'RootComponents'),
+                        path.join(moduleDir, 'src', 'RootComponents'),
+                        path.join(moduleDir, 'lib', 'RootComponents')
+                    ],
+                    []
+                ),
+                context
+            }),
+            new webpack.EnvironmentPlugin(projectConfig.env),
+            new ServiceWorkerPlugin({
+                mode,
+                paths,
+                injectManifest: true,
+                injectManifestConfig: {
+                    include: [/\.js$/],
+                    swSrc: './src/sw.js',
+                    swDest: './sw.js'
+                }
+            }),
+            new UpwardIncludePlugin({
+                upwardDirs: hasFlag('upward')
+            }),
```

Comment:
I was kind of expecting us not to include the `UpwardIncludePlugin` at all if `hasFlag('upward')` is `false`. 
Does the plugin fail gracefully without any `upwardDirs`?

Reply:
@supernova-at Good catch! We should always require an upward file in the app itself.